packages/flutter/lib/src/services/mouse_cursor_android.dart

```diff
@@ -0,0 +1,117 @@
+import 'package:flutter/gestures.dart';
+
+import 'platform_channel.dart';
+
+// The channel interface with the platform.
+//
+// It is separated to a class for the conventience of reference by the shell
+// implementation.
+@immutable
+class _AndroidPlatformActions {
+  const _AndroidPlatformActions(this.mouseCursorChannel);
+
+  final MethodChannel mouseCursorChannel;
+
+  // Set cursor as a sytem cursor specified by `platformConstant`.
+  Future<void> setAsSystemCursor({
+    @required int platformConstant,
+  }) {
+    assert(platformConstant != null);
+    return mouseCursorChannel.invokeMethod<void>(
+      'setAsSystemCursor',
+      <String, dynamic>{
+        'platformConstant': platformConstant,
+      },
+    );
+  }
+}
+
+/// Controls mouse cursor behaviors on an
+/// [Android](https://developer.android.com/reference)
+/// shell over a method channel.
+///
+/// See also:
+///
+///  * [StandardMouseCursorManager], which uses this class on the Android
+///    platform.
+class MouseCursorAndroidDelegate extends MouseCursorPlatformDelegate {
+  /// Create a [MouseCursorAndroidDelegate] by providing the method channel to use.
+  ///
+  /// The [mouseCursorChannel] must not be null, and is usually
+  /// [SystemChannels.mouseCursor].
+  MouseCursorAndroidDelegate({@required MethodChannel mouseCursorChannel})
+    : assert(mouseCursorChannel != null),
+      _platform = _AndroidPlatformActions(mouseCursorChannel);
+
+  // System cursor constants are used to set system cursor on Android.
+  // Must be kept in sync with Android's [PointerIcon#Constants](https://developer.android.com/reference/android/view/PointerIcon.html#constants_2)
+
+  /// The same constant as Android's `PointerIcon.TYPE_DEFAULT`,
+  /// used internally to set system cursor.
+  static const int kPlatformConstantDefault = 1000;
+
+  /// The same constant as Android's `PointerIcon.TYPE_ARROW`,
+  /// used internally to set system cursor.
+  static const int kPlatformConstantArrow = 1000;
+
+  /// The same constant as Android's `PointerIcon.TYPE_GRAB`,
+  /// used internally to set system cursor.
+  static const int kPlatformConstantGrab = 1020;
+
+  /// The same constant as Android's `PointerIcon.TYPE_GRABBING`,
+  /// used internally to set system cursor.
+  static const int kPlatformConstantGrabbing = 1021;
+
+  /// The same constant as Android's `PointerIcon.TYPE_HAND`,
+  /// used internally to set system cursor.
+  static const int kPlatformConstantHand = 1002;
+
+  /// The same constant as Android's `PointerIcon.TYPE_NULL`,
+  /// used internally to set system cursor.
+  static const int kPlatformConstantNull = 0;
+
+  /// The same constant as Android's `PointerIcon.TYPE_TEXT`,
+  /// used internally to set system cursor.
+  static const int kPlatformConstantText = 1008;
+
+  final _AndroidPlatformActions _platform;
+
+  Future<bool> _activatePlatformConstant(int platformConstant) async {
+    await _platform.setAsSystemCursor(platformConstant: platformConstant);
```

Comment:
Why the indirection via _AndroidPlatformActions? Couldn't this method just implement that method call directly?